app/src/androidTest/java/com/cmput301f20t21/bookfriends/ui/login/LoginActivityTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2020. Book Friends Team
+ * All rights reserved.
+ * github URL: https://github.com/CMPUT301F20T21/Book_Friends
+ */
+
+package com.cmput301f20t21.bookfriends.ui.login;
+
+
+import androidx.test.espresso.ViewInteraction;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.LargeTest;
+
+import com.cmput301f20t21.bookfriends.R;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.replaceText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.allOf;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class LoginActivityTest {
+
+    @Rule
+    public ActivityScenarioRule<LoginActivity> mActivityTestRule = new ActivityScenarioRule<>(LoginActivity.class);
+
+
+    @Test
+    public void testLoginSuccess() throws InterruptedException {
+        ViewInteraction usernameField = onView(
+                allOf(withId(R.id.login_username_field), isDisplayed()));
+        usernameField.perform(replaceText("test"), closeSoftKeyboard());
```

Comment:
I wonder if we need to mock the service class when we test stuffs that rely on the database. For example if I as an admin delete this user from the db, this test would fail. We could discuss more about this for tonight's meeting.